corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -59,6 +59,35 @@ hqDefine("data_dictionary/js/data_dictionary", [
             self.deprecated(false);
         };
 
+        self.showAllowedValues = function () {
+            console.log('show allowed values UI here!');
+            // I attempted to use something like this but wasn't able to wrangle HQ's JS system to work.
+            // some docs are here: https://commcare-hq.readthedocs.io/js-guide/code-organization.html
```

Comment:
There are some doc updates in progress, so I'd start at the top level of the js guide: https://github.com/dimagi/commcare-hq/tree/master/docs/js-guide#dimagi-javascript-guide - though the examples above are likely to be more helpful than the general js docs.